### ISSUE TITLE
V2.3.0-RC3 ManagedAtmosphereHandler:message discard the second request on the same method

### ISSUE THREAD

**Issue by dreamershl** [Atmosphere/atmosphere#1880]
When request is sent from client to server. The @Message annotated method will be invoked by the ManagedAtmosphereHandler:message() accordingly. But the following check will discard the second request to the same method!!! 

Another point, personally feel the function process isn't consistent. In the following code, it directly treats the input parameter as the return result. While the other part, will process that input parameter using "Invoker.invokeMethod" first.  
    
     Method invokedMethod = (Method) request.getAttribute(getClass().getName());
            request.removeAttribute(getClass().getName());

            if (invokedMethod != null) {
                for (MethodInfo m : onRuntimeMethod) {
                    if (invokedMethod.equals(m.method)) {
                        return m.encode(encoders, o);
                    }
                }
            }

**Comment by jfarcand** [comment]
@dreamershl Which version worked for you? Can you elaborate on the issue?

**Comment by dreamershl** [comment]
It seems the issue exists from beginning. I find this issue is after I decided to combine the @Message methods into one in order to avoid the exception based method choosing (based on my JSON parsing experience, the exception approach is extremely slow because of thread context lock). 

The following is my current message handler. It means all the client request will be all redirected to this one method. Then I realize the server always miss out the second request. It is because the second request to this method is considered as processed request by the ManagedAtmosphereHandler:message() and it will not pass the request to the handler anymore.  

    @Message(encoders = {ProtoMsgEncoder.class}) @DeliverTo(DeliverTo.DELIVER_TO.RESOURCE)
                       public @Nullable StockProto.StockMsgPack onMessage(AtmosphereResource r, Object origMsg)
    {
            if (origMsg instanceof String)
            {
                //request from the client side directly
                ...
            }
            else if (origMsg instanceof StockProto.StockMsgPack)
            {
                //server side broadcasting
                ...
            }
            else if (origMsg instanceof List<?>)
            {
                //atmosphere framework cached message for disconnected clients
                List<?> msgList = (List<?>) origMsg;
               ....
            }

        return retMsgPack;
    }

**Comment by jfarcand** [comment]
Fix for #1887 fix the current issue as well. @dreamershl Please test and let me know the result. Also, can you elaborate on the last part of your issue:
```
Another point, personally feel the function process isn't consistent. In the following code, it directly treats the input parameter as the return result. While the other part, will process that input parameter using "Invoker.invokeMethod" first.
```

**Comment by jfarcand** [comment]
Fixed by https://github.com/Atmosphere/atmosphere/commit/91cbe7e8647c7f152d3ec456e1fe5d03dca74d79
